### ISSUE TITLE
Linq.Dynamic and EntityFramework.Extensions

### ISSUE THREAD

**Issue by robc-centralpower** [zzzprojects/EntityFramework.Extended#114]
There is a conflict with .OrderBy from Linq.Dynamic and the EntityFramework.Extensions.

I can't use both libraries because of ambiguous error on the .OrderBy function.

If you are going to include Linq.Dynamic library in EntityFramework.Extensions can you make the functions publicly accessible.

I use the Linq.Dynamic .OrderBy function with ordering an IQueryable by a dynamic string parameter.

Thanks.

**Comment by mstancombe** [comment]
Have also been bitten by this.  Have issued a pull request to change the namespace of the internally used linq.dynamic so it is prefixed by the assembly namespace and wont conflict with referenced usable of the dynamic linq lib.

**Comment by nap7276** [comment]
+1 Same issue.

**Comment by damiangreen** [comment]
+1

**Comment by ninety7** [comment]
+1 Any plans on fixing this?

**Comment by rykkart** [comment]
+1

**Comment by ismcagdas** [comment]
+1

**Comment by samerzmd** [comment]
Come on man look at profile picture that's my face now

**Comment by runxc1** [comment]
I am running into the same problem.... probably a good idea to add a dependency rather than include it

**Comment by neekgreen** [comment]
+1

**Comment by J-ohn** [comment]
+1

**Comment by karriku** [comment]
+1

**Comment by aeslinger0** [comment]
+1 Two years and no movement on this :( One workaround is to right-click the reference in the project, select properties, change the "Aliases" property from "**global**" to something like "**EFExtended**", and then include the following at the top of the class you want to use this library in:

```
extern alias EFExtended;
using EFExtended.EntityFramework.Extensions;
```

**Comment by theeheng** [comment]
Thank [aeslinger0](https://github.com/aeslinger0), you save my day.

**Comment by aeslinger0** [comment]
No problem. I actually still had problems using it along side Linq.Dynamic in the same classes so I ended up making a class within my own namespace that wraps the methods from the framework, using the extern reference in the class, so I could then use the class anywhere along side Linq.Dynamic without any problems.

**Comment by samiraguiar** [comment]
Is there any update on this?
@pwelter34

**Comment by JonathanMagnan** [comment]
Hello @samiraguiar ,

EntityFramework.Extended has not been supported since 2015. I highly recommend you to move to [Entity Framework Plus](http://entityframework-plus.net/) which is also free, open source and supported!

There is currently no plan to support this library but if you have the same issue with EF+, I will be happy to fix it.

Looking at some answer, there is currently some workaround if you still want to continue to use EntityFramework.Extended.

Let me know if that answered correctly to your question.

Best Regards,

Jonathan

**Comment by BStrat77** [comment]
Thank you @aeslinger0 , you saved the day for me as well...years later.

For other people dealing with this, when I upgraded my projects to use the NuGet PackageReference format this issue appeared again. It doesn't look like there is a way to easily alias in the Properties like before, at least that I could find, so I had to do the following in the csproj:

`<Target Name="ChangeAliasesOfNugetRefs" 
    BeforeTargets="FindReferenceAssembliesForReferences;ResolveReferences">
    <ItemGroup>
      <ReferencePath Condition="'%(FileName)' == 'EntityFramework.Extended'">
        <Aliases>EFExtended</Aliases>
      </ReferencePath>
    </ItemGroup>
</Target>`

(ref: https://stackoverflow.com/a/46484086/1442323)

I did not have to do an alias previous to the PackageReference format.
